### PR TITLE
Add documentation for test plugin command

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

Add documentation for tanzu test command to test plugins

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

make docs-lint
Screenshot 2023-04-06 at 17 36 44

### Release note

```release-note

Add documentation for tanzu test command to test plugins
```

### CODE REVIEW

File: docs/plugindev/README.md [hunk 1]
```diff
@@ -259,6 +259,40 @@ Note: as the configuration variable implies, the
 plugin development and testing only. The use of this setting in the production setting
 is not supported.
 
+### Testing the plugins
+
+Plugin tests can be run by installing the admin `test` plugin.
+Currently, we only support testing plugins build on local.
```
Comment (anujc25) [inline docs/plugindev/README.md:265]:
```suggestion
Currently, we only support testing plugins built locally.
```

File: docs/plugindev/README.md [hunk 2]
```diff
@@ -259,6 +259,40 @@ Note: as the configuration variable implies, the
...
+
```
Comment (anujc25) [inline docs/plugindev/README.md:266]: How about adding a note which explains the plugin is deprecated and we are not planning to enhance this in the future? cc @marckhouzam @vuil @mpanchajanya 

```suggestion

The `test` admin functionality has been deprecated and no future enhancements are planned for this plugin.

```

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks for adding this.

File: docs/plugindev/README.md (hunk 2 above)
Comment (marckhouzam) [reply docs/plugindev/README.md:268]: I think that is a good idea.

**Pull request merged**